pennylane/ops/op_math/pow.py

```diff
@@ -329,7 +329,7 @@ def generator(self):
 
         See also :func:`~.generator`
         """
-        return self.z * self.base.generator()
+        return qml.s_prod(self.z, self.base.generator()).simplify()
```

Comment:
How come simplify is necessary here?

Reply:
Without simplifying, we have the following possible case:

```
op = qml.RX(0.2, [0])
op = qml.pow(op, 5)
op.generator()
```
Which prints:
```
5 * -0.5 * X(0)
```
That is, if the argument of `pow` is another scalar product, the multiplication between the two scalars is not performed by default. I initially separated the case where the argument is a `SProd` instance, but I switched to `simplify` after discovering such a function.

Reply:
Instead of simplifying, you could use `lazy=False`:
```suggestion
        return qml.s_prod(self.z, self.base.generator(), lazy=False)
```

This will make it so that if the base is also an `SProd`, the coefficients will be merged into one.